cellrank/external/kernels/_wot_kernel.py

```diff
@@ -499,6 +508,59 @@ def _generate_cost_matrices(
             f"`{type(cost_matrices).__name__}` is not yet implemented."
         )
 
+    def _threshold_transition_matrix(self, threshold: float, method: str) -> None:
+        tmat = self._orig_tmat.copy()
+        print("Using method", method)
+        if method == "perc-local":
+            assert 0 <= threshold <= 100
+            for i in range(tmat.shape[0]):
+                start, end = tmat.indptr[i], tmat.indptr[i + 1]
+                data = tmat.data[start:end]
+                t = np.percentile(data, threshold)
+                data[data <= t] = 0
+                print(t, data.shape, np.sum(data <= t), np.sum(data))
+                tmat.data[start:end] = data
+        elif method == "perc-global":
+            assert 0 <= threshold <= 100
+            threshold = np.percentile(tmat.data, threshold)
+            print("Threshold:", threshold)
+            tmat.data[tmat.data <= threshold] = 0.0
+        elif method == "fixed":
+            tmat.data[tmat.data <= threshold] = 0.0
+        elif method == "k":
+            assert isinstance(threshold, int)
+            assert 0 <= threshold <= tmat.shape[1]
+            for i in range(tmat.shape[0]):
+                start, end = tmat.indptr[i], tmat.indptr[i + 1]
+                data = tmat.data[start:end]
+                ixs = np.argsort(data)[::-1]
+                data[ixs[threshold:]] = 0
+                tmat.data[start:end] = data
+        elif method == "auto":
+            minn = np.inf
+            for i in range(tmat.shape[0]):
```

Comment:
Why not use list comprehension with a single `min`? `max` with `map`, converting to a list and then taking the minimum could potentially be even quicker but I doubt it (due to the list conversion)

Reply:
Done.